app/validators/routing/new_when_rule_validator.py

```diff
@@ -0,0 +1,196 @@
+from app.validators.routing.types import (
+    python_type_to_json_type,
+    resolve_value_source_json_type,
+    TYPE_DATE,
+    TYPE_ARRAY,
+    TYPE_NUMBER,
+    TYPE_STRING,
+    TYPE_NULL,
+    TYPE_BOOLEAN,
+)
+from app.validators.validator import Validator
+
+OPERATOR_NOT = "not"
+OPERATOR_AND = "and"
+OPERATOR_OR = "or"
+OPERATOR_EQUAL = "=="
+OPERATOR_NOT_EQUAL = "!="
+OPERATOR_GREATER_THAN = ">"
+OPERATOR_LESS_THAN = "<"
+OPERATOR_GREATER_THAN_OR_EQUAL = ">="
+OPERATOR_LESS_THAN_OR_EQUAL = "<="
+OPERATOR_IN = "in"
+OPERATOR_ALL_IN = "all-in"
+OPERATOR_ANY_IN = "any-in"
+OPERATOR_DATE = "date"
+
+BOOLEAN_OPERATORS = [OPERATOR_NOT, OPERATOR_AND, OPERATOR_OR]
+
+COMPARISON_OPERATORS = [
+    OPERATOR_EQUAL,
+    OPERATOR_NOT_EQUAL,
+    OPERATOR_GREATER_THAN,
+    OPERATOR_LESS_THAN,
+    OPERATOR_GREATER_THAN_OR_EQUAL,
+    OPERATOR_LESS_THAN_OR_EQUAL,
+]
+
+ARRAY_OPERATORS = [OPERATOR_IN, OPERATOR_ALL_IN, OPERATOR_ANY_IN]
```

Comment:
I wonder if this this should have 'in' in the var somewhere, down if the _if conditions_ I had to come back up to check what the array operators included to immediate think all the _ins_

Reply:
I named it this to make it clear that these are operators that work with arrays. I don't think we can combine `ARRAY` with `IN` in the name as the order implies something i.e. `ARRAY_IN` vs `IN_ARRAY`. We could name it `IN_OPERATORS`.

Reply:
IN_OPERATORS for me

Reply:
I was going to make the change, but after trying it I'm not convinced. As with the rename of `boolean` to `logic`, if we rename this we should also rename `array` to `in` in the schemas; this leads to paths of `in/in.json`, which I think is less clear then `array/in.json`